CoreUtilities/src/au/gov/asd/tac/constellation/utilities/gui/filechooser/FileChooser.java

```diff
@@ -0,0 +1,126 @@
+        } else {
+            // This is not the UI thread so just call the open dialog method
+            return CompletableFuture.completedFuture(
+                    openFileDialog(fileChooserBuilder, FileChooserMode.SAVE)
+            );
+        }
+    }
+    
+    /**
+     * Construct a file chooser using the passed builder and opens it as a open
+     * dialog.
+     * <p/>
+     * If this was called from a UI thread then this will wrap the request
+     * in a async future call that does a recursive call back to this method. This
+     * ensures that the UI thread does not get into a race condition where the
+     * UI thread becomes blocked.
+     * 
+     * @param fileChooserBuilder the file chooser builder to build and display
+     * @return a {@link CompletableFuture} that will return the selected file or
+     *     null if the user selects cancel
+     */
+    public static CompletableFuture<File> openOpenDialog(final FileChooserBuilder fileChooserBuilder) {
+        if (SwingUtilities.isEventDispatchThread() || Platform.isFxApplicationThread()) {
+            // Wrap the open call in a completable future so this happens on another
+            // thread and return immediately
+            return CompletableFuture.supplyAsync(() ->
+                    openFileDialog(fileChooserBuilder, FileChooserMode.OPEN));
+        } else {
+            // This is not the UI thread so just call the open dialog method
+            return CompletableFuture.completedFuture(
+                    openFileDialog(fileChooserBuilder, FileChooserMode.OPEN)
+            );
+        }
+    }
+    
+    /**
+     * Construct a file chooser using the passed builder and opens it as a open
+     * or save dialog depending on the passed mode. This method will block the
+     * calling thread until the user selects a file or presses cancel.
+     *
+     * @param fileChooserBuilder the file chooser builder to build and display
+     * @param fileDialogMode the type of file chooser dialog to open, save or open
+     * @return the selected file or null if the user selects cancel
+     */
+    private static File openFileDialog(final FileChooserBuilder fileChooserBuilder,
+                                       final FileChooserMode fileDialogMode) {
+        final ShowFileChooserDialog showDialog = new ShowFileChooserDialog(
+                fileChooserBuilder, fileDialogMode);
+        
+        try {
+            // Make a request to open the file chooser dialog on the UI thread
+            EventQueue.invokeAndWait(showDialog);
+        } catch (InterruptedException ex) {
+            LOGGER.log(Level.WARNING, "Thread displaying the file chooser was interrupted.", ex);
+            Thread.currentThread().interrupt();
+        } catch (InvocationTargetException ex) {
+            LOGGER.log(Level.SEVERE, "Error occured during selection in file chooser.", ex);
+            Exceptions.printStackTrace(ex);
```

Comment:
can you remove this stack trace print?

Reply:
Done

Reply:
I believe the throwable parameter in the logger will do this (from my recollection)